docs/reference/setup/logging-config.asciidoc

```diff
@@ -194,3 +196,36 @@ files (four rolled logs, and the active log).
 
 You can disable it in the `config/log4j2.properties` file by setting the deprecation
 log level to `error`.
+
+
+[float]
+[[json-logging]]
+=== Json log format
+
+To make parsing elasticsearch logs easier, logs are now printed in a json format.
+This is configured by a log4j layout property `appender.rolling.layout.type = ESJsonLayout`.
+This layout requires a type_name attribute to be set which is used to distinguish
+logs streams when parsing.
+[source,properties]
+--------------------------------------------------
+appender.rolling.layout.type = ESJsonLayout
+appender.rolling.layout.type_name = server
+--------------------------------------------------
+
+Each line contains a single json with set of attributes that are configured in `ESJsonLayout`.
+See this class javadoc for more details.
```

Comment:
I am not sure we should point users of our reference docs to the Javadoc. In the end it would mean that they'd either need to checkout the sources or go to Github, find the right class and read the Javadocs which seems inconvenient. I wonder whether we should go the extra mile and document the properties here as well although the information is duplicated then. Or we don't document it in Javadoc and only in the reference docs. Either way, I think we can improve this

Reply:
or maybe I could point them directly?
We do in sam documentation pages already. For isntance: https://www.elastic.co/guide/en/elasticsearch/client/java-rest/6.0/java-rest-high-query-builders.html